main.py

```diff
@@ -27,17 +38,45 @@
             # get the full path to the file *after* the base reproschema_folder path
             # since files can be referenced by relative paths, we need to keep track of relative location
             filename = str(file.relative_to(reproschema_folder))
-            reproschema_content[filename] = load_file(file)
+            with open(f"{reproschema_folder}/{filename}") as f:
+                reproschema_content[filename] = json.loads(f.read())
+
+    schema_name = [
+        name for name in list(reproschema_content.keys())
```

Comment:
when doing a list comprehension, you don't need to wrap the generator in `list()`, because it will iterate over the generator.

if you want a list of the keys, then yes you do need to do `key_list = list(reproschema_content.keys())`, but you don't need it if you are doing `key_list = [k for k in reproschema_content.keys()]`, because your list comprehension is already iterating over the generator (whereas before, you needed to call `list()` to do that). hope that makes sense, useful to know in general